Futur Protocol

morgueye4 opened the pull request:
# Milestone Delivery Checklist

- [x] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [X] This pull request is being made by the same account as the accepted application.
- [X] I have disclosed any and all sources of reused code in the submitted repositories and have done my due diligence to meet its license requirements.
- [X] In case of acceptance, invoices must be submitted and payments will be transferred to the Polkadot AssetHub and/or fiat account provided in the application.
- [X] The delivery is according to the [Guidelines for Milestone Deliverables](https://grants.web3.foundation/docs/Support%20Docs/milestone-deliverables-guidelines).

Link to the application pull request: https://github.com/w3f/Grants-Program/pull/2168 < please fill this in with the PR number of your application.

semuelle commented on deliveries/FuturProtocol_milestone_1.md:
```diff
@@ -0,0 +1,29 @@
+
+**The delivery is according to the official [milestone delivery guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/milestone-deliverables-guidelines.md).**  
+
+* **Application Document:** [Futur Protocol](https://github.com/RELAI-Network/Grants-Program/blob/fd99d75e580c84560d8099cb404d4b10b48295b4/applications/FuturFusion.md)
```
```suggestion
* **Application Document:** [Futur Protocol](https://github.com/w3f/Grants-Program/blob/master/applications/FuturFusion.md)
```

keeganquigley reviewed (changes requested):
Thanks for the delivery @morgueye4 

- the link in delivery "4. Futur Console" gives [404 error](https://github.com/RELAI-Network/futur-console-react), can you fix this?
- the "3. SAST/DAST module" deliverable links to the [Mobsf](https://mobsf.github.io/docs), but I'm not sure what is supposed to be evaluated here. Am I supposed to be running something to coordinate with backend and off-chain workers?

Thanks for any further instructions you can provide!

keeganquigley commented:
Also are there any unit tests for the [backend functions](https://github.com/RELAI-Network/relai-fn)?

morgueye4 commented:
Thank you @keeganquigley for the feedback.

> the link in delivery "4. Futur Console" gives 404 error, can you fix this?

We omitted  to turn the [repository](https://github.com/RELAI-Network/futur-console-react) public it was private. Now it is accessible.

> the "3. SAST/DAST module" deliverable links to the Mobsf, but I'm not sure what is supposed to be evaluated here. Am I supposed to be running something to coordinate with backend and off-chain workers?

Here the Deliverable is a deployed instance of [MobSF](https://github.com/MobSF/Mobile-Security-Framework-MobSF) in a server in a GCP instance with the [proxy](https://github.com/RELAI-Network/sast-dast-gateway) alongside with the MobSF instance.

You can run it locally by using Docker by following the [instructions](https://mobsf.github.io/docs/#/mobsf_docker).

In the context of the FuturConsole, our instance is called by the Frontend Console [futur-console-react](https://github.com/RELAI-Network/futur-console-react) to submit apks or to scan and retrieve results

> Also are there any unit tests for the [backend functions](https://github.com/RELAI-Network/relai-fn)?

I have updated the readme and added a [guide](Also are there any unit tests for the [backend functions](https://github.com/RELAI-Network/relai-fn)?) on how to test the functions with the Firebase emulator.

keeganquigley commented:
Thanks @morgueye4 for the heads up. Are there any other repos that are private? Please [take note of our policy](https://github.com/w3f/grants-program?tab=readme-ov-file#guidelines) when it comes to open-sourcing.

morgueye4 commented:
> Thanks @morgueye4 for the heads up. Are there any other repos that are private? Please [take note of our policy](https://github.com/w3f/grants-program?tab=readme-ov-file#guidelines) when it comes to open-sourcing.

Welcome @keeganquigley and thanks for sharing but no worries, there are no other private repos, this one was an omission about turning it public.

keeganquigley reviewed (changes requested):
Thanks @morgueye4 I was able to get through everything and am now just in the last steps of setting up my developer account at https://console-relai.web.app/developer

It is asking me for some $RL to submit the tx. Is this the testnet token? Can you please send some to 5H1SgrToMp1RYhbUjzQmVxpRGRvo3s2j4xKUjCWJ9GQAVREM

Thanks! Also, I expected to get an insufficient balance error but instead it just hangs:

<img width="660" alt="relai" src="https://github.com/w3f/Grant-Milestone-Delivery/assets/35080151/ef5f73bf-80c3-4267-a04f-6dec012fc56f">

morgueye4 commented:
> Thanks @morgueye4 I was able to get through everything and am now just in the last steps of setting up my developer account at https://console-relai.web.app/developer
> 
> It is asking me for some $RL to submit the tx. Is this the testnet token? Can you please send some to 5H1SgrToMp1RYhbUjzQmVxpRGRvo3s2j4xKUjCWJ9GQAVREM
> 
> Thanks! Also, I expected to get an insufficient balance error but instead it just hangs:
> 
> <img alt="relai" width="660" src="https://private-user-images.githubusercontent.com/35080151/322986252-ef5f73bf-80c3-4267-a04f-6dec012fc56f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMzMDA0NzQsIm5iZiI6MTcxMzMwMDE3NCwicGF0aCI6Ii8zNTA4MDE1MS8zMjI5ODYyNTItZWY1ZjczYmYtODBjMy00MjY3LWEwNGYtNmRlYzAxMmZjNTZmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDE2VDIwNDI1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVkODUyYjc1NDc0NzZjZWI0NzA2ZjM3MGJhNDY2NzBlYTVlNTU5ZjBmMTA5YThmMmU1MmFiODkwNDEwOWQwNDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.RGv3eyoVZDrWohQs1a4k-9An4jgyzQ6_FdFUyKT7Yak">

Welcome @keeganquigley ! I just sent you some test tokens at the address you provided ! I have also updated the account creation [guide](https://blog.relai.network/futur-console-account-creation-relai-network-devnet) for token requesting for new accounts.
We will have a look at the error handling for low balance and the like cases.

keeganquigley commented:
Awesome thanks @morgueye4 would be much appreciated. I was able to get it to work and am now willing to pass the milestone, here is my [final evaluation](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/futur_1_keeganquigley.md). Congrats!

morgueye4 commented:
> Awesome thanks @morgueye4 would be much appreciated. I was able to get it to work and am now willing to pass the milestone, here is my [final evaluation](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/futur_1_keeganquigley.md). Congrats!

Thank you @keeganquigley and the team for giving a chance !

RouvenP commented:
hi @morgueye4 we sent the payment last Friday
